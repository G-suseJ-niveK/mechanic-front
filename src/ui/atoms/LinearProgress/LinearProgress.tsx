import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import MaterialLinearProgress from '@material-ui/core/LinearProgress';

const useStyles = makeStyles(() => ({
  colorPrimary: {
    backgroundColor: '#b2dfdb'
  }
}));

type LinearProgressProps = {
  loading: boolean;
};

const LinearProgress: React.FC<LinearProgressProps> = (props: LinearProgressProps) => {
  const classes = useStyles();
  const { loading } = props;

  return <>{loading && <MaterialLinearProgress classes={classes} />}</>;
};

export default LinearProgress;
