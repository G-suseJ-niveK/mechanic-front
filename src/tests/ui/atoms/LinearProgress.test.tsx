import React from 'react';
import { render, screen } from '@testing-library/react';
import LinearProgress from '~atoms/LinearProgress/LinearProgress';

describe('Render LinearProgress', () => {
  test('should render is not loading', () => {
    const { container } = render(<LinearProgress loading={false} />);

    expect(container).toMatchSnapshot();
  });

  test('should render is loading', () => {
    render(<LinearProgress loading={true} />);

    expect(screen.getByRole('progressbar')).toBeInstanceOf(Node);
  });
});
