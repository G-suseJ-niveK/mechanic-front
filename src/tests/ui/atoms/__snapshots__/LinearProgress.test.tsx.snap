// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render LinearProgress should render is not loading 1`] = `<div />`;
